packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/simulationservice/interventions/Intervention.java

```diff
@@ -32,4 +34,36 @@ public Intervention clone() {
 		}
 		return intervention;
 	}
+
+	// Check for no negative timesteps
+	// Check for no duplicate time + appliedTo pairs in static interventions
+	public Boolean validateIntervention() throws Exception {
+		for (int i = 0; i < this.staticInterventions.size(); i++) {
+			final Number time = this.staticInterventions.get(i).getTimestep();
+			if (time.doubleValue() < 0) {
+				final String errorMessage = String.format(
+					"The intervention %s has a timestep %s which is less than 0.",
+					this.getName(),
+					time.toString()
+				);
+				throw new Exception(errorMessage);
+			}
+			for (int j = 0; j < this.staticInterventions.size(); j++) {
```

Comment:
Right now this validation is O(n^4).  I think you can simplify it a bit doing something like...

```
if(staticInterventions.size() > 1){
   for(int i = 1; i < staticInterventions.size(); i++){
      ...
      	final String appliedToOne = this.staticInterventions.get(i).getAppliedTo();
				final Number timeOne = this.staticInterventions.get(i).getTimestep();
				final String appliedToTwo = this.staticInterventions.get(i-1).getAppliedTo();
				final Number timeTwo = this.staticInterventions.get(i-1).getTimestep();
   }
}
```

This would let you get rid of your `if` statement as well because `appliedToTwo` would never be `appliedToOne`